### ISSUE TITLE
Client.Post returns an error when needs to unpack a time.Time field

### ISSUE THREAD

**Issue by grzn** [99designs/gqlgen#1372]
The error returned is
```
expected a map, got 'string'
```

mapstructure doesn't support decoding natively, there's a need for adding a custom decoder, see https://github.com/mitchellh/mapstructure/issues/159

Can you please add it?

**Comment by jozuenoon** [comment]
Yes, that would be very useful, also maybe passing custom decode hooks?

**Comment by frederikhors** [comment]
> there's a need for adding a custom decoder

Thanks @grzn, can I ask you to post your function, please? I'm new to Golang reflect pkg.

Thanks, really.

**Comment by rsergeychik** [comment]
I've got the same problem in my project where I had mapstructure v1.1.2 as indirect dependency. It seems that v1.x.x overrides v0.x.x and does not work correctly. Seems like upgrading mapstructure dependency so that it's no v0 should fix that.
